src/core/grid_based_algorithms/lbgpu.cpp

```diff
@@ -32,6 +32,7 @@
 #include "debug.hpp"
 #include "global.hpp"
 #include "grid.hpp"
+#include "grid_based_algorithms/lb.hpp"
```

Comment:
you are now introducing a dependency of the GPU LB on the CPU LB... maybe move the check to ```lb_interface.hpp```